apps/src/templates/instructions.jsx

```diff
@@ -0,0 +1,59 @@
+var msg = require('../locale');
+
+module.exports = React.createClass({
+
+  propTypes: {
+    puzzleTitle: React.PropTypes.string,
+    instructions: React.PropTypes.string,
+    instructions2: React.PropTypes.string,
+    renderedMarkdown: React.PropTypes.string,
+    authoredHints: React.PropTypes.array,
+    markdownClassicMargins: React.PropTypes.bool,
+    aniGifURL: React.PropTypes.string
+  },
+
+  render: function () {
+
+    // The logic here needs some serious cleanup
+    var body;
+    if (this.props.renderedMarkdown) {
+      body = (<div 
+        className={ this.props.markdownClassicMargins ? "instructions-markdown classic-margins" : "instructions-markdown" }
```

Comment:
Not sure if it's possible/realistic, but if these classnames are used here and only here, it would be great if we could use inline styles instead of class names.